src/plugins/smith_waterman/src/sw_cuda_cpp.cpp

```diff
@@ -176,34 +178,58 @@ QList<resType> calculateOnGPU(const char *seqLib, int seqLibLength, ScoreType *q
 
     //************************** declare arrays on device
 
-    char *g_seqLib;
-    ScoreType *g_queryProfile;
-    ScoreType *g_HdataMax;
-    ScoreType *g_HdataUp;
-    ScoreType *g_HdataRec;
+    char      *g_seqLib           = nullptr;
+    ScoreType *g_queryProfile     = nullptr;
+    ScoreType *g_HdataMax         = nullptr;
+    ScoreType *g_HdataUp          = nullptr;
+    ScoreType *g_HdataRec         = nullptr;
     ScoreType *g_HdataTmp;
-    ScoreType *g_FdataUp;
-    ScoreType *g_directionsUp;
-    ScoreType *g_directionsMax;
-    ScoreType *g_directionsRec;
-    int *g_directionsMatrix = nullptr;
-    int *g_backtraceBegins = nullptr;
+    ScoreType *g_FdataUp          = nullptr;
+    ScoreType *g_directionsUp     = nullptr;
+    ScoreType *g_directionsMax    = nullptr;
+    ScoreType *g_directionsRec    = nullptr;
+    int       *g_directionsMatrix = nullptr;
+    int       *g_backtraceBegins  = nullptr;
 
     //************************** allocate global memory on device
 
-    cudaMalloc((void **)&g_seqLib, sizeL);
-    cudaMalloc((void **)&g_queryProfile, sizeP);
-    cudaMalloc((void **)&g_HdataMax, sizeQ);
-    cudaMalloc((void **)&g_HdataUp, sizeQ);
-    cudaMalloc((void **)&g_FdataUp, sizeQ);
-    cudaMalloc((void **)&g_directionsUp, sizeQ);
-    cudaMalloc((void **)&g_directionsMax, sizeQ);
-    cudaMalloc((void **)&g_HdataRec, sizeQ);
-    cudaMalloc((void **)&g_directionsRec, sizeQ);
+    QList<cudaError> errors = { cudaMalloc((void **)&g_seqLib, sizeL),
+                                cudaMalloc((void **)&g_queryProfile, sizeP),
+                                cudaMalloc((void **)&g_HdataMax, sizeQ),
+                                cudaMalloc((void **)&g_HdataUp, sizeQ),
+                                cudaMalloc((void **)&g_FdataUp, sizeQ),
+                                cudaMalloc((void **)&g_directionsUp, sizeQ),
+                                cudaMalloc((void **)&g_directionsMax, sizeQ),
+                                cudaMalloc((void **)&g_HdataRec, sizeQ),
+                                cudaMalloc((void **)&g_directionsRec, sizeQ) };
 
     if (U2::SmithWatermanSettings::MULTIPLE_ALIGNMENT == resultView) {
         cudaError errorMatrix = cudaMalloc(reinterpret_cast<void **>(&g_directionsMatrix), directionMatrixSize);
         cudaError errorBacktrace = cudaMalloc(reinterpret_cast<void **>(&g_backtraceBegins), backtraceBeginsSize);
+        errors << errorMatrix << errorBacktrace;
+    }
+
+    if (std::find_if(errors.constBegin(), errors.constEnd(), [](const cudaError error) {
```

Comment:
nit: remove const from param or use const type& to make it more reasonable (we do not use method-local const across the code)